PSReadLine/Movement.cs

```diff
@@ -90,25 +102,53 @@ private void MoveToLine(int numericArg)
                         : point.X;
             }
 
-            var topLine = _initialY;
+            int newCurrent;
+            if (_moveToLineDesiredColumn == endOfLine)
+            {
+                newCurrent = _current;
+
+                if (numericArg > 0)
+                {
+                    // Moving to a subsequent line.
+                    for (int i = 0; i < numericArg; i++)
+                    {
+                        for (newCurrent++; newCurrent < _buffer.Length && _buffer[newCurrent] != '\n'; newCurrent++) ;
```

Comment:
Does buffer hold physical lines, which themselves can be multiple logical lines if they contain `\n`?

If so, is there a danger arrow-up will move up a logical line at the end of a line, but a physical one everywhere else?

I'm suspicious because of how different the logic for moving up a line is now when the cursor is at the end of a line, compared to this being either a small edge-case catch or a broader change that deals with logical vs physical lines everywhere

Reply:
`_buffer` holds the user input text, something like `fff\nfffffff\nffff` for the following input:
```
PS> fff
>> fffffff
>> ffff
```
I'm not sure I fully understand your comment, but the behavior I think it should have is:
1. if the cursor is **NOT** at the end of a logical line, then <kbd>Up</kbd> and <kbd>Down</kbd> move the cursor up/down 1 or N **physical lines** (depends on `numericArg`). This is the existing behavior.
2. if the cursor is at the end of a logical line, then <kbd>Up</kbd> and <kbd>Down</kbd> move the cursor up/down 1 or N **logical lines** (depends on `numericArg`), and cursor is always put at the end of the new logical line.
Let's sync tomorrow to understand your concerns.